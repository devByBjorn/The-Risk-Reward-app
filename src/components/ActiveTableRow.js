import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
// import { calculatePositiveR, calculateNegativeR } from '../calculations/riskRewardCalculation'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteFirebaseTrade } from '../actions/tradeActions'

const ActiveTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    entry,
    stop,
    target,
    status,
    opened,
    negativeR,
    positiveR,
    id }) => (
    status === 'active' &&
    <tr>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{entry}</td>
      <td>{stop}</td>
      <td>{target}</td>
      <td>{opened}</td>
      <td>{negativeR}</td> {/* loss -R*/}
      <td>{positiveR}</td> {/* win +R*/}
      <td>
        <div>
          <Link to={`/edit-trade/${id}`}>
            <EditIcon />
          </Link>
          <button onClick={() => {
            dispatch(deleteFirebaseTrade({ id }))
          }}
          ><TrashIcon />
          </button>
        </div>
      </td>
    </tr>
  )

export default connect()(ActiveTableRow)