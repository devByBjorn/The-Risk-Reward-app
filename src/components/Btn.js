import React from 'react'

export const Btn = (props) => (
  <button
    type="button"
    {...props}
  >
    {props.text}</button>
)

export const SubmitBtn = (props) => (
  <button>{props.text}</button>
)

export default Btn