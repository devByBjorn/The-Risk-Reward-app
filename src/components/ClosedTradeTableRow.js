import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { EditIcon, TrashIcon } from '../icons/IconsComponents'
import { deleteTrade } from '../actions/tradeActions'




const ClosedTradeTableRow = (
  {
    dispatch,
    market,
    direction,
    setup,
    opened,
    closed,
    status,
    outcome,
    rewardToRisk,
    id }) => {

  const [rowColor, setRowColor] = useState('')

  const colors = {
    win: '#7FBA00',
    loss: '#F0341B',
    scratch: '#FFB900'
  }
  const { win, loss, scratch } = colors

  useEffect(() => {
    if (outcome === 'win') {
      setRowColor(win)
    } else if (outcome === 'loss') {
      setRowColor(loss)
    } else {
      setRowColor(scratch)
    }
  })

  return (
    status === 'closed' &&

    <tr style={{ backgroundColor: rowColor }}>
      <td>{market}</td>
      <td>{direction}</td>
      <td>{setup}</td>
      <td>{rewardToRisk}</td>
      <td className="desktop">{opened}</td>
      <td className="desktop">{closed}</td>
      <td className="desktop">{outcome}</td>
      <td className="handle-trade-row">
        <div className="handle-trade">
          <Link to={`/edit-trade/${id}`}>
            <EditIcon
              className="edit-icon icon"
            />
          </Link>
          <button
            className="btn-transparent"
            onClick={() => {
              alert('Do you want to delete this trade?')
              dispatch(deleteTrade({ id }))
            }}
          ><TrashIcon
              className="trashcan-icon icon"
            />
          </button>
        </div>
      </td>
    </tr>
  )
}
export default connect()(ClosedTradeTableRow)