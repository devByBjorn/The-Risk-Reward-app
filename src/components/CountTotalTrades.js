import React from 'react'
import { connect } from 'react-redux'

const ClosedTradesTotal = ({ trades }) => (
  <h2>Total Closed Trades: {trades.length}</h2>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades.filter((trade) => trade.status === 'closed')
  }
}

export default connect(mapStateToProps)(ClosedTradesTotal)