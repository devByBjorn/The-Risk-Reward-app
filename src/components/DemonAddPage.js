import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { addDemon } from '../actions/demonActions'
import DemonFinderForm from './demon-finder/DemonFinderForm'

const AddDemonPage = (props) => (
  <div>
    <h1>Add Demon Page</h1>
    <NavLink to="/demons">Back to demon list</NavLink>
    <div>
      <ol>
        <li>Give your demon suiting a title</li>
        <li>Think about your demon in terms of what,when,why,  how.
          <ul>
            <li>What makes the demon take over?</li>
            <li>When does this demon take over?</li>
            <li>Why...</li>
            <li>How can you manage this demon?</li>
          </ul>
        </li>
        <li>Write down your thoughts and process of managing this demon</li>
      </ol>
      <div>
        <h2>Demon</h2>
        <DemonFinderForm
          handleSubmit={(demon) => {
            props.dispatch(addDemon(demon))
            props.history.push('/demons')
          }}
        />
      </div>
    </div>
  </div>
)

export default connect()(AddDemonPage)