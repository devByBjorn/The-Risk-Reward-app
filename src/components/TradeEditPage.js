import React, { Fragment } from 'react'
import { connect } from 'react-redux'
import { editFirebaseTrade, deleteFirebaseTrade } from '../actions/tradeActions'
import FormParent from './FormParent'
//import FormParentTwo from './FormParentTwo'

export class TradeEditPage extends React.Component {
  handleEditOnAdd = (trade) => {
    this.props.editFirebaseTrade(this.props.trade.id, trade)
    this.props.history.push('/trades')
  }
  render() {
    return (
      <Fragment>
        <FormParent
          //pathname={this.props.location.pathname}
          trade={this.props.trade}
          handleSubmit={this.handleEditOnAdd}
        />
      </Fragment>
    )
  }
}

const mapStateToProps = (state, props) => ({
  trade: state.trades.find((trade) => trade.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  editFirebaseTrade: (id, trade) => dispatch(editFirebaseTrade(id, trade)),
  deleteFirebaseTrade: (id) => dispatch(deleteFirebaseTrade(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(TradeEditPage)

