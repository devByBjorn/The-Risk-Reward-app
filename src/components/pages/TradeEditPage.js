import React from 'react'
import { connect } from 'react-redux'
import TradeTableFormComplete from '../forms/TradeTableFormComplete'
import { editTrade, deleteTrade } from '../../actions/tradeActions'
import { TrashIcon } from '../../icons/IconsComponents'

class TradeEditPage extends React.Component {
  handleEditOnAdd = (trade) => {
    this.props.editTrade(this.props.trade.id, trade)
    this.props.history.push('/trades')
  }
  handleDeleteOnTrash = () => {
    this.props.deleteTrade({ id: this.props.trade.id })
    this.props.history.push('/trades')
  }
  render() {
    return (
      <div>
        <TradeTableFormComplete
          trade={this.props.trade}
          handleSubmit={this.handleEditOnAdd}
        />
        <button
          onClick={this.handleDeleteOnTrash}>
          <TrashIcon /> Trash trade</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    trade: state.trades.find((trade) => trade.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch) => ({
  editTrade: (id, trade) => dispatch(editTrade(id, trade)),
  deleteTrade: (id) => dispatch(deleteTrade(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(TradeEditPage)