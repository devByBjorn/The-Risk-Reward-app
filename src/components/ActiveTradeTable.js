import React from 'react'
import { connect } from 'react-redux'
import ActiveTableRow from './ActiveTableRow'
import { ThMobileAndDesktop, ThOnlyDesktop } from './TradeTableHead'
const ActiveTradeTable = ({ trades }) => (
  <div>
    <table className="table-grid active">
      <caption>Active Trades</caption>
      <thead>
        <tr>
          <ThMobileAndDesktop heading='Market' />
          <ThMobileAndDesktop heading='Direction' />
          <ThMobileAndDesktop heading='Setup' />
          <ThMobileAndDesktop heading='Entry' />
          <ThMobileAndDesktop heading='Stop' />
          <ThMobileAndDesktop heading='Target' />
          <ThOnlyDesktop heading='Opened' />
          <ThMobileAndDesktop heading='-R' />
          <ThMobileAndDesktop heading='+R' />
        </tr>
      </thead>
      <tbody>
        {trades.map((trade) =>
          (<ActiveTableRow
            {...trade}
            key={trade.id}
          />)
        )}
      </tbody>
    </table>
  </div>
)

const mapStateToProps = (state) => {
  return {
    trades: state.trades,
  }
}

export default connect(mapStateToProps)(ActiveTradeTable)