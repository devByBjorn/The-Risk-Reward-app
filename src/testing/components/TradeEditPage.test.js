import React from 'react'
import { shallow } from 'enzyme'
import { TradeEditPage } from '../../components/TradeEditPage'
import { trades } from '../fixtures/trades'

let editFirebaseTrade, deleteFirebaseTrade, history, wrapper

beforeEach(() => {
  editFirebaseTrade = jest.fn()
  deleteFirebaseTrade = jest.fn()
  history = { push: jest.fn() }
  wrapper = shallow(
    <TradeEditPage
      editFirebaseTrade={editFirebaseTrade}
      deleteFirebaseTrade={deleteFirebaseTrade}
      history={history}
      trade={trades[0]}
    />
  )
})

test('should render TradeEditPage', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle editFirebaseTrade', () => {
  wrapper.find('FormParent').prop('handleSubmit')(trades[0])
  expect(history.push).toHaveBeenLastCalledWith('/trades')
  expect(editFirebaseTrade).toHaveBeenLastCalledWith(trades[0].id, trades[0])
})

