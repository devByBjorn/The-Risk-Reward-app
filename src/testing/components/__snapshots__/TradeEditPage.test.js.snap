// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TradeEditPage 1`] = `
<Fragment>
  <FormParent
    handleSubmit={[Function]}
    trade={
      Object {
        "closed": 1587115800000,
        "conclusion": Object {
          "execution": "abc",
          "improveExecution": "def",
          "improveManagement": "jkl",
          "management": "ghi",
          "whyExecution": "mno",
          "whyManagement": "pqr",
        },
        "direction": "long",
        "entry": 120,
        "exit": 140,
        "id": "123",
        "market": "dax",
        "opened": 1587031851000,
        "outcome": "win",
        "period": 0,
        "rMultiple": 2,
        "reward": false,
        "rewardToRisk": 2,
        "risk": false,
        "setup": "ibl tick under",
        "status": "closed",
        "stop": 110,
        "target": 140,
      }
    }
  />
</Fragment>
`;
